src/renderer/modals/Terms/index.js

```diff
@@ -26,73 +26,97 @@ const TermsModal = () => {
   const [accepted, setAccepted] = useState(false);
   const onSwitchAccept = useCallback(() => setAccepted(!accepted), [accepted]);
 
-  const onClick = useCallback(() => {
-    acceptTerms();
+  const onClickClose = useCallback(() => {
     dispatch(closeModal("MODAL_TERMS"));
   }, [dispatch]);
 
+  const onClick = useCallback(() => {
+    acceptTerms();
+    onClickClose();
+  }, [onClickClose]);
+
   return (
-    <Modal name="MODAL_TERMS" preventBackdropClick centered>
-      <ModalBody
-        title={<Trans i18nKey="Terms.title" />}
-        render={() => (
-          <>
-            <TrackPage category="Modal" name="Terms" />
+    <Modal
+      name="MODAL_TERMS"
+      preventBackdropClick
+      centered
+      render={(
+        { data }: { data?: { showClose: boolean } }, // Why u no get the correct type ?
```

Comment:
Hi! :wave: i'm pretty sure you can retrieve data you send during the from dispatch event right from the component props.
I updated it not so long ago for the props to be accessible on the entry point of each modals have a look into `src/renderer/ModalsLayer.js`

You can then retrieve show close this way:
```javascript
const TermsModal = ({ name, showClose }: { name: string, showClose: boolean }) => {

const onClickClose = useCallback(() => {
    dispatch(closeModal(name));
  }, [dispatch]);
}

return (
    <Modal
      name={name} ...
```

Reply:
will try that